### PR TITLE
Fix Ordered List on Delete

### LINKED ISSUE #21: Handle Child Node Being Deleted
Right now this breaks ordered proxy chains.

### PR DESCRIPTION
**Pull request by tpendragon**
Closes #21

This uses the proxy's interface rather than a service object. It's moderately complex, if it's too much so I can just override delete_records to do all this and we can support that.

### CODE REVIEW

**Comment by terrellt** [comment]
Adding unit tests to this now.

**Comment by jcoyne** [comment]
Please squash these commits. :smile:

File: lib/active_fedora/aggregation/association.rb [hunk 1]
```diff
@@ -5,14 +5,8 @@ def ordered_reader
       OrderedReader.new(owner).to_a
     end
 
-    def add_link(proxy)
-      LinkInserter.new(owner, proxy).call
-    end
-
-    def save_through_record(record)
```
Comment (jcoyne) [inline lib/active_fedora/aggregation/association.rb]: This is called by ActiveFedora::Associations::IndirectlyContainsAssociation#insert_record(). Are you sure you have test coverage for insert_record()?

Comment (terrellt) [inline lib/active_fedora/aggregation/association.rb]: Yeah, this just doesn't override that behavior it's inheriting.

**Pull request merged**